File: Aio-Algorithms-And-Solutions/Solutions/HackerRank/RecursiveDigitSum/RecursiveDigitSum.java
package HackerRank.RecursiveDigitSum;

public class RecursiveDigitSum {
    public static int superDigit(String n, int k)
    {
        long sum = 0;

        for (int i = 0; i < n.length(); ++i)
            sum += Integer.parseInt(String.valueOf(n.charAt(i)));

        sum *= k;

        if (sum > 9)
            return superDigit(String.valueOf(sum), 1);
        else
            return (int)sum;

    }
}

File: Aio-Algorithms-And-Solutions/Solutions/HackerRank/RecursiveDigitSum/RecursiveDigitSum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleTestSolutions.HackerRank.RecursiveDigitSum
{
    class RecursiveDigitSum
    {
        public static int superDigit(string n, int k)
        {
            long sum = 0;
                                     
            for (int i = 0; i < n.Length; ++i) 
                sum += Int32.Parse(n[i].ToString());

            sum *= k;

            if (sum > 9) 
                return superDigit(sum.ToString(), 1);
            else 
                return (int)sum;
        }
    }
}
